### ISSUE TITLE
no-undefined-types: false positive for @template with @typedef

### ISSUE THREAD

**Issue by kevinoid** [gajus/eslint-plugin-jsdoc#579]
## Expected behavior

The code below would not produce any warnings.

## Actual behavior

    /tmp/example.js
      5:0  warning  The type 'ValueType' is undefined  jsdoc/no-undefined-types
      5:0  warning  The type 'ValueType' is undefined  jsdoc/no-undefined-types
    
    ✖ 2 problems (0 errors, 2 warnings)

## ESLint Config

```json
{
  "extends": [
    "plugin:jsdoc/recommended"
  ],
  "settings": {
    "jsdoc": {
      "mode": "typescript"
    }
  }
}
```

## ESLint sample

```js
/**
 * A function returning the same type as its argument.
 *
 * @template ValueType
 * @typedef {function(ValueType): ValueType} ValueFunc
 */
```

## Environment

* Node version: v14.0.0
* ESLint version v7.2.0
* `eslint-plugin-jsdoc` version: 27.0.7

**Comment by kevinoid** [comment]
After further consideration, the example code should probably be:

```js
/**
 * A function returning the same type as its argument.
 *
 * @template ValueType
 * @typedef {function(ValueType): ValueType} ValueFunc<ValueType>
 */
```

since `ValueFunc` should have a type parameter.  It produces the same warnings as the original example, but causes an additional `no-undefined-types` when the type is used, e.g.:

```js
/**
 * @template ValueType
 * @param {ValueFunc<ValueType>} valueFunc Function to run.
 */
function run(valueFunc) {
}
```

I'm unsure how `@template` and `@typedef` are supposed to be used together.  My apologies if this is user error.
